.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,113 @@
+        sed 's/^\([^;]*;[^;]*\);.*$/\1/' base/unicodetools/data/ucd/dev/UnicodeData.txt > base-repertoire.txt
+        if diff base-repertoire.txt merged-repertoire.txt
+        then echo "repertoire-changed=false" >> "$GITHUB_OUTPUT"
+        else echo "repertoire-changed=true" >> "$GITHUB_OUTPUT"
+        fi
+    - name: Check pipeline labels
+      id: check-labels
+      if: steps.compare-repertoire.outputs.repertoire-changed == 'true'
+      run: |
+        python3 <<EOF
+        import os, json
+        with open(os.environ['GITHUB_EVENT_PATH'], 'r') as f:
+          event = json.load(f)
+        print(event)
+        labels = event['pull_request']['labels']
+        pipeline_labels = [label for label in labels if label['name'].startswith('pipeline-')]
+        if not pipeline_labels:
+          print("::error title=Missing pipeline label::PRs for character additions must have a pipeline label.")
+          exit(1)
+        if len(pipeline_labels) > 1:
+          print("::error title=Multiple pipeline labels::Only one pipeline-* label must be applied.")
+          exit(1)
+        label = pipeline_labels[0]
+        print("Labeled", label)
+        with open(os.environ['GITHUB_OUTPUT'], 'a') as f:
+          print("pipeline-label=" + label['name'], file=f)
+        EOF
+  l2-document:
+    needs: labels-for-repertoire-changes
+    if: ${{ always() && needs.labels-for-repertoire-changes.outputs.repertoire-changed == 'true' }}
+    name: Proposal document
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check L2 document
+      run: |
+        python3 <<EOF
+        import os, json, re
+        with open(os.environ['GITHUB_EVENT_PATH'], 'r') as f:
+          event = json.load(f)
+        print(event)
+        pr_body = event['pull_request']['body']
+        if not re.search(r"L2/\d\d-\d\d\d", pr_body):
+          print("::error title=Need proposal document::PRs for character additions must include a link to an L2 document in the PR description.")
+          exit(1)
+        EOF
+  utc-decision:
+    needs: labels-for-repertoire-changes
+    if: needs.labels-for-repertoire-changes.outputs.pipeline-label != 'pipeline-recommended-to-UTC'
+    name: UTC decision
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check UTC decision
+      run: |
+        python3 <<EOF
+        import os, json, re
+        with open(os.environ['GITHUB_EVENT_PATH'], 'r') as f:
+          event = json.load(f)
+        print(event)
+        pr_body = event['pull_request']['body']
+        if not re.search(r"UTC-\d\d\d-[MC]\d", pr_body):
+          print("::error title=Need UTC decision::PRs for approved or provisionally assigned must include a link to a UTC decision.")
+          exit(1)
+        EOF
+  draft-unless-approved:
+    needs: labels-for-repertoire-changes
+    if: needs.labels-for-repertoire-changes.outputs.pipeline-label != 'pipeline-16.0'
```

Comment:
`16` should be a variable, perhaps a repo variable <https://github.com/unicode-org/unicodetools/settings/variables/actions>

Reply:
don't we have `16.0` somewhere?  The python script could discover this. Take the `if` out and analyze it within the python script